[OSDOCS-3901] Created the topic map for ROSA HCP (#55565)
openshift/openshift-docs: modules/rosa-hcp-vpc-terraform.adoc

// Module included in the following assemblies:
//
// * rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-quickly.adoc
// * rosa_getting_started/quickstart.adoc

:_content-type: PROCEDURE
[id="rosa-hcp-vpc-terraform_{context}"]
= Creating an AWS VPC using Terraform

Terraform is a tool that allows you to create various resources using an established template. The following process uses the default options as required to create a {hcp} for ROSA cluster. For more information about using Terraform, see the additional resources.

.Prerequisites

* You have installed Terraform on your machine.

.Procedure

. In your CLI, create a directory for your terraform files, and then navigate to that directory.
+
[source,terminal]
----

> $ mkdir hypershift-tf && cd hypershift-tf

Reviewer: ```suggestion
$ mkdir hypershift-tf
$ cd hypershift-tf
```

----

. Download the VPC setup terraform file.

> //This is Will's personal one, so we may need to create one for GA

Reviewer: ```suggestion
```

Updated to use a more generalized template, we will _not_ be providing a terraform option for GA (as far as I'm aware)

Author: Could documentation create a template for the sake of standardizing our docs? I'm not totally certain of the logistics of that, but it could be very helpful!

+
[source,terminal]
----

> $ curl -s -o setup-vpc.tf https://raw.githubusercontent.com/openshift-cs/OpenShift-Troubleshooting-Templates/master/rosa-hcp-terraform/setup-vpc.tf

Reviewer 2: Could we have this terraform file on a Red Hat controlled domain?

----

. After the template has downloaded, initiate the terraform file. You see a message confirming the initialization when this process completes. 
+
[source,terminal]
----
$ terraform init
----

. Run the `plan` command to build your VPC terrafom plan based off of the downloaded template. You may specify a cluster name and/or your AWS region.
+
[source,terminal]
----
$ terraform plan -out rosa.plan [-var aws_region=<region>] [-var cluster_name=<cluster_name>]
----

. You should have a `rosa.plan` file in the directory that you created in the first step. Apply this plan file to build your VPC.
+
[source,terminal]
----
$ terraform apply rosa.plan
----

> . Optional: You can set the values of the private, public, and machinepool subnets that were created as part of the terraform process as variables to use when creating your {hcp} for ROSA cluster.

Reviewer: ```suggestion
. Optional: You can capture the values of the terraform provisioned private, public, and machinepool subnet IDs  as environment variables to use when creating your {hcp} for ROSA cluster.
```

// Module included in the following assemblies:
//
// * rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-quickly.adoc
// * rosa_getting_started/quickstart.adoc

:_content-type: PROCEDURE
[id="rosa-hcp-vpc-terraform_{context}"]
= Creating an AWS Virtual Private Cloud using Terraform

Terraform is a tool that allows you to create various resources using an established template. The following process uses the default options as required to create a {hcp} for ROSA cluster. For more information about using Terraform, see the additional resources.

.Prerequisites

* You have installed Terraform on your machine.

.Procedure

> . Open a shell prompt and create a directory for your terraform files by running the following command:

Reviewer 3: ```suggestion
. Open a shell prompt and create a directory for your Terraform files by running the following command:
```

+
[source,terminal]
----
$ mkdir hypershift-tf
----

. Navigate to the created directory by running the following command:
+
[source,terminal]
----
$ cd hypershift-tf
----

> . Download the VPC setup terraform file by running the following command:

Reviewer 3: ```suggestion
. Download the VPC setup Terraform file by running the following command:
```

+
[source,terminal]
----
$ curl -s -o setup-vpc.tf https://raw.githubusercontent.com/openshift-cs/OpenShift-Troubleshooting-Templates/master/rosa-hcp-terraform/setup-vpc.tf
----

> . After the template has downloaded, initiate the terraform file by running the following command. You see a message confirming the initialization when this process completes: 
> +
> [source,terminal]
> ----
> $ terraform init
> ----

Reviewer 3: ```suggestion
. After the template has downloaded, initiate the Terraform file by running the following command: 
+
[source,terminal]
----
$ terraform init
----
+
A message confirming the initialization appears when this process completes.
```

> . To build your VPC terrafom plan based off of the downloaded template, run the `plan` command. You may specify a cluster name and your AWS region.

Reviewer 3: ```suggestion
. To build your VPC Terrafom plan based off of the downloaded template, run the `plan` command. You can specify a cluster name and your AWS region.
```

+
[source,terminal]
----
$ terraform plan -out rosa.plan [-var aws_region=<region>] [-var cluster_name=<cluster_name>]
----

. You should have a `rosa.plan` file in the directory that you created in the first step. Apply this plan file to build your VPC by running the following command:
+
[source,terminal]
----
$ terraform apply rosa.plan
----

> . Optional: You can capture the values of the terraform-provisioned private, public, and machinepool subnet IDs as environment variables to use when creating your {hcp} for ROSA cluster by running the following command:
> +
> [source,terminal]
> ----
> $ PRIVATE_SUBNET=`terraform output -raw cluster-private-subnet`
> $ PUBLIC_SUBNET=`terraform output -raw cluster-public-subnet`
> $ MACHINEPOOL_SUBNET=`terraform output -raw node-private-subnet`
> ----

Reviewer 3: Technically, there should be three substeps here (only one command is allowed in one step). Something like:

```suggestion
. Optional: You can capture the values of the Terraform-provisioned private, public, and machinepool subnet IDs as environment variables to use when creating your {hcp} for ROSA cluster by running the following commands:

** For the private subnet ID:
+
[source,terminal]
----
$ PRIVATE_SUBNET=`terraform output -raw cluster-private-subnet`
----

** For the public subnet ID:
+
[source,terminal]
----
$ PUBLIC_SUBNET=`terraform output -raw cluster-public-subnet`
----

** For the machinepool subnet ID:
+
[source,terminal]
----
$ MACHINEPOOL_SUBNET=`terraform output -raw node-private-subnet`
----
```
